client/jetpack-cloud/sections/partner-portal/hooks.tsx

```diff
@@ -280,6 +280,80 @@ export function useLicenseIssuing(
 	return [ issue, isLoading ];
 }
 
+const useAssignLicenses = (
+	licenseKeys: Array< string >,
+	selectedSite: { ID: number; domain: string } | null
+) => {
+	const products = useProductsQuery( {
+		select: selectAlphaticallySortedProductOptions,
+	} );
+	const dispatch = useDispatch();
+	const fromDashboard = getQueryArg( window.location.href, 'source' ) === 'dashboard';
+	const assignLicense = useAssignLicenseMutation( {
+		onError: ( error: Error ) => {
+			dispatch( errorNotice( error.message, { isPersistent: true } ) );
+		},
+	} );
+	const isLoading = assignLicense.isLoading;
+	const selectedSiteId = selectedSite?.ID as number;
+	const assignMultipleLIcenses = useCallback( async () => {
+		const assignLicenseRequests: any = [];
+		licenseKeys.forEach( ( licenseKey ) => {
+			dispatch(
+				recordTracksEvent( 'calypso_partner_portal_assign_license_submit', {
+					license_key: licenseKey,
```

Comment:
How about we do an event tracking for all the `licenseKeys` at once rather than doing it for individual licenses?

```
dispatch(
			recordTracksEvent( 'calypso_partner_portal_assign_multiple_licenses_submit', {
				products: licenseKeys.join( ',' ),
			} )
);
```